api/src/middlewares/services/ses.ts

```diff
@@ -0,0 +1,121 @@
+import logger, { Logger } from "pino";
+import config from "config";
+import {
+  GetTemplateCommand,
+  SendRawEmailCommand,
+  SESClient,
+  Template,
+} from "@aws-sdk/client-ses";
+import { ERRORS } from "../../errors";
+import { createMimeMessage } from "mimetext";
+import {
+  AffirmationTemplateData,
+  CNITemplateData,
+} from "../../handlers/helpers/getTemplateDataFromInputs";
+import { ses } from "../../SESClient";
+
+export class SESService {
+  templates: {
+    cni?: string;
+    affirmation?: string;
+  } = {};
+  logger: Logger;
+  ses: SESClient;
+
+  constructor() {
+    this.templates.affirmation = config.get("affirmationTemplate");
+    this.templates.cni = config.get("cniTemplate");
+    this.logger = logger().child({ service: "SES" });
+    this.ses = ses;
+  }
+
+  async getTemplate(type: "cni" | "affirmation") {
+    try {
+      if (!this.templates[type]) {
+        this.logger.error(ERRORS.ses.NO_TEMPLATE);
+        return;
+      }
+      const templateData = await this.ses.send(
+        new GetTemplateCommand({
+          TemplateName: this.templates[type],
+        })
+      );
+
+      if (!templateData?.Template) {
+        this.logger.error(ERRORS.ses.TEMPLATE_NOT_FOUND);
+        return;
+      }
```

Comment:
looking at docs.. I don't think it actually returns anything undefined - it will throw if something doesn't return. Do you mind double checking if that's the case?